pixeltable/dataframe.py

```diff
@@ -119,46 +118,116 @@ def finalize_exec(self) -> None:
             self.filter.release()
 
 
+
 class DataFrame:
     def __init__(
             self, tbl: catalog.TableVersion,
-            select_list: Optional[List[exprs.Expr]] = None,
+            select_list:  Optional[List[Tuple[exprs.Expr, Optional[str]]]] = None,
             where_clause: Optional[exprs.Predicate] = None,
             group_by_clause: Optional[List[exprs.Expr]] = None,
             order_by_clause: Optional[List[Tuple[exprs.Expr, bool]]] = None,  # List[(expr, asc)]
             limit: Optional[int] = None,):
         self.tbl = tbl
+
+        # select list logic
+        DataFrame._select_list_check_rep(select_list) # check select list without expansion
         # exprs contain execution state and therefore cannot be shared
-        self.select_list = copy.deepcopy(select_list)  # None: implies all cols
+        select_list = copy.deepcopy(select_list)
+        exps, column_names = DataFrame._normalize_select_list(tbl, select_list)
+        DataFrame._select_list_check_rep(list(zip(exps, column_names))) # check select list after expansion to catch early
+        self._exps = exps
```

Comment:
You're basically caching something here, and it can deviate from the actual self.select_list (which might be none).

Choose a more meaningful name (select_list_exprs?) and prefix with a comment that these two are always set, even when select_list is None (which is an important part of their semantics).